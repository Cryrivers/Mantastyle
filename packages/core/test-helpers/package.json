{
  "name": "@manta-style/test-helpers",
  "version": "0.2.0-alpha.9",
  "main": "src/index.ts",
  "dependencies": {
    "prettier": "^1.14.3"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@types/prettier": "^1.13.2"
  }
}
