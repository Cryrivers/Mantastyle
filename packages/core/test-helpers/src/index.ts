import * as BuilderPluginTestHelper from './builder-tester';
export { BuilderPluginTestHelper };
