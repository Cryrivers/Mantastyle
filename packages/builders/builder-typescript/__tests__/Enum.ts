import getTranspiledString from '../src/utils/transpile';
import { BuilderPluginTestHelper } from '@manta-style/test-helpers';

const testTranspiledString = BuilderPluginTestHelper.testTranspiledString(
  getTranspiledString,
);

describe('Enums', () => {
  test('Normal Enums', () => {
    expect(testTranspiledString('enum Test { A, B, C, D };')).toMatchSnapshot();
  });
  test('Enums with numeric initializer', () => {
    expect(
      testTranspiledString('enum Test { A, B = 12, C, D };'),
    ).toMatchSnapshot();
  });
  test('Enums with string initializer', () => {
    expect(
      testTranspiledString("enum Test { A = 'A', B = 'B', C = 'C', D = 'D' };"),
    ).toMatchSnapshot();
  });
});
