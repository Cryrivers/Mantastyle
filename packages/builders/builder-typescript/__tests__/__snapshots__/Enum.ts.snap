// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enums Enums with numeric initializer 1`] = `
"FROM:
enum Test { A, B = 12, C, D };

--------------------------------------------------------
TO:
'use strict';
exports.__esModule = true;
var runtime_1 = require('@manta-style/runtime');
var Test = runtime_1['default'].EnumDeclaration({ A: 0, B: 12, C: 13, D: 14 });
"
`;

exports[`Enums Enums with string initializer 1`] = `
"FROM:
enum Test { A = 'A', B = 'B', C = 'C', D = 'D' };

--------------------------------------------------------
TO:
'use strict';
exports.__esModule = true;
var runtime_1 = require('@manta-style/runtime');
var Test = runtime_1['default'].EnumDeclaration({
  A: 'A',
  B: 'B',
  C: 'C',
  D: 'D',
});
"
`;

exports[`Enums Normal Enums 1`] = `
"FROM:
enum Test { A, B, C, D };

--------------------------------------------------------
TO:
'use strict';
exports.__esModule = true;
var runtime_1 = require('@manta-style/runtime');
var Test = runtime_1['default'].EnumDeclaration({ A: 0, B: 1, C: 2, D: 3 });
"
`;
